tw-tasks-core/src/main/java/com/transferwise/tasks/triggering/KafkaTasksExecutionTriggerer.java

```diff
@@ -284,7 +288,11 @@ public void poll(String bucketId) {
           mdcService.with(() -> {
             mdcService.put(task);
 
-            TaskTriggering taskTriggering = new TaskTriggering().setTask(task).setBucketId(bucketId).setOffset(offset)
+            TaskTriggering taskTriggering = new TaskTriggering()
+                .setTask(task)
+                .setBucketId(bucketId)
+                .setOffset(offset)
+                .setTriggerAt(Instant.ofEpochMilli(consumerRecord.timestamp()))
```

Comment:
While not clear in the docs this long appears to be epoch time in millis

https://kafka.apache.org/21/javadoc/org/apache/kafka/clients/consumer/ConsumerRecord.html#timestamp--

https://kafka.apache.org/10/javadoc/org/apache/kafka/connect/data/Timestamp.html